pkg/lclient/fakeclient.go

```diff
@@ -143,15 +192,14 @@ func (m *mockDockerClient) ContainerRemove(ctx context.Context, containerID stri
 }
 
 func (m *mockDockerClient) ContainerInspect(ctx context.Context, containerID string) (types.ContainerJSON, error) {
-	for _, containerElement := range mockContainerList {
+	for _, containerElement := range mockContainerJSONList {
 		if containerElement.ID == containerID {
 			containerConfig := container.Config{
-				Image:  containerElement.Image,
-				Labels: containerElement.Labels,
```

Comment:
After removing labels, the unit tests will fail and the ContainerJson doesn't have a labels field. To fix this we have to do a few changes

Change https://github.com/openshift/odo/blob/cd7f174bec9f488065889073760bd76289c409dc/pkg/url/url.go#L173
to 
```
remoteURLName := containers[i].Labels[internalPort.Port()]
```
Here `i` is the loop counter variable. We retrieved the container label value from the container struct directly since they have the same id.

The other change is https://github.com/openshift/odo/blob/cd7f174bec9f488065889073760bd76289c409dc/pkg/url/url.go#L575
to 
```
dockerURL := getMachineReadableFormatDocker(internalPort.Int(), externalport, portbinding[0].HostIP, containers[i].Labels[internalPort.Port()])
 ```